### ISSUE TITLE
False positives: Snippet props and CSS custom properties

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#72]
### Error Breakdown (v0.6.6)

| Error Code | Count | Description |
|------------|-------|-------------|
| TS2322 | 123 | Snippet type not assignable |
| TS2769 | 52 | No overload matches this call |
| TS2353 | 2 | CSS custom properties not in props type |
| TS2321 | 1 | Excessive stack depth with Plugin type |

### Reproductions

https://github.com/janosh/svelte-check-rs-repro

#### 1. CSS Custom Properties (TS2353)

```svelte
<MyComponent --point-fill-color="green" />
```

Error: `'--point-fill-color' does not exist in type '__SvelteProps_...'`

#### 2. Snippet Props (TS2322, TS2769)

```svelte
<MyComponent>
  {#snippet tooltip({ x, y })}
    <div>Point at ({x}, {y})</div>
  {/snippet}
</MyComponent>
```

Error: `Type '() => { '{@render ...} must be called with a Snippet': ... }' is not assignable to type 'Snippet<[...]>'`

#### 3. Vite Plugin Types (TS2321)

```ts
{ name: `custom`, enforce: `pre`, load(id) { ... } } satisfies Plugin
```

Error: `Excessive stack depth comparing types '...' and 'Plugin<any>'`

### Reproduce

```bash
git clone https://github.com/janosh/svelte-check-rs-repro
pnpm install && pnpm check:rs
```

For all 178 errors: https://github.com/janosh/matterviz
